docs/When_Do_You_Need_A_Custom_Device.md

```diff
@@ -0,0 +1,73 @@
+2. Is a hardware driver available?<br />
+3. Is the driver well documented?<br />
+
+VeriStand uses channels to pass data between different parts of the system, including to and from custom devices. All VeriStand channels are LabVIEW double data type (DBL). See **[LabVIEW Help](https://zone.ni.com/reference/en-XX/help/371361R-01/)** » **[Fundamentals](https://zone.ni.com/reference/en-XX/help/371361R-01/TOC5.htm)** » **[Building the Block Diagram](https://zone.ni.com/reference/en-XX/help/371361R-01/TOC12.htm)** » **[How-To](https://zone.ni.com/reference/en-XX/help/371361R-01/TOC14.htm)** » **[Floating Point Numbers](https://zone.ni.com/reference/en-XX/help/371361R-01/lvhowto/floating_point_numbers/)** for more information on LabVIEW data types.
+ 
+Can the hardware requirement be met by passing LabVIEW DBLs to and from the custom device during steady state operation?
+If the hardware driver returns a vector, structure, or any non-DBL data, it cannot be passed directly from the custom device to the rest of the VeriStand system. The developer is responsible for coercing the data (or using an alternative communication mechanism) to pass data from the custom device to the rest of the system. For more information on the available communication mechanisms, see **[LabVIEW 2018 Real-Time Module Help](https://zone.ni.com/reference/en-XX/help/370715P-01/)** » **[Real-Time Module Concepts](https://zone.ni.com/reference/en-XX/help/370715P-01/lvrtconcepts/rtmoduleconcepts/)** » **[Sharing Data in Deterministic Applications](https://zone.ni.com/reference/en-XX/help/370715P-01/lvrtconcepts/rtsharedatadeterm/)** » **[Exploring Remote Communication Methods](https://zone.ni.com/reference/en-XX/help/370715P-01/lvrtconcepts/exploring_communication_methods/)**.
+VeriStand also exposes its TCP pipe via dynamic event registration. This pipe may suite your remote communication requirements. See the Custom Device Engine Events section for more information.
+
+#### Testing
+
+A custom device is one part of a VeriStand system. The complete state of the operator's system is seldom known by the custom device developer. System state includes the following information.
+1 What are the specifications of the Target and Host Computer? Check the **[Glossary](https://www.ni.com/documentation/en/veristand/latest/manual/veristand-glossary/)** section of the **[VeriStand Help](https://www.ni.com/documentation/en/veristand/latest/manual/manual-overview/)**.
+2 What components are in the system definition?
+3 How computationally intense are the simulation models?
+4 What loop rates are required?
+5 What is the health and resource utilization of the system?
+Ideally, the custom device is implemented to be minimally burdensome, extremely efficient, and easy to use. Depending on its complexity, it may become necessary to test, debug, and optimize the code on systems representative of the operator’s system. Consider the following example.
```

Comment:
```suggestion

Ideally, the custom device is implemented to be minimally burdensome, extremely efficient, and easy to use. Depending on its complexity, it may become necessary to test, debug, and optimize the code on systems representative of the operator’s system.

```